src/types.ts

```diff
@@ -25,3 +25,12 @@ export type BaseProps = PresentationalProps & { unstyled?: boolean };
  * valid `<svg>` element attribute as props.
  */
 export type IconComponent = FunctionComponent<JSX.SVGAttributes<SVGSVGElement>>;
+
+/**
+ * A component type used by other components that can optionally support a
+ * transition animation
+ */
```

Comment:
These components will still be usable as standalone components. e.g. 

```jsx
<Slider visible>
  <div>Some sort of content</div>
</Slider>
```

should totally work.

As such, perhaps:

```suggestion
/**
 * A component that describes an `in` and an `out` transition, typically to animate the 
 * mounting and unmounting of a child component.
 */
```

I dunno, this one is hard to write!